### PR TITLE
support AMP for In-batch Negatives

### PR DESCRIPTION
**Pull request by tianxin1860**
### PR types

New features
### PR changes

Models
### Description

1. support AMP for In-batch Negatives
2. AMP 加速数据: 在 batch_size=64, max_seq_length=64 条件下，v100 单卡加速比约 1.94
```
FP32 训练: 3.11 steps/s
AMP 训练: 6.04 steps/s

测试条件如下: 
1. V100 单卡
2. batch_size=64
2. max_seq_length=64
3. paddlepaddle-gpu==2.2.0
```

### CODE REVIEW

**Comment by ZeyuChen** [comment]
是否可以补充下性能提升数据是否符合预期？

**Comment by tianxin1860** [comment]
> 是否可以补充下性能提升数据是否符合预期？

已补充 AMP 加速数据: 在 batch_size=64, max_seq_length=64 条件下，v100 单卡加速比约 1.94, 加速性能符合预期。

File: examples/semantic_indexing/train_batch_neg.py [hunk 1]
```diff
@@ -133,17 +135,30 @@ def do_train():
...
+                    args.use_amp,
+                    custom_white_list=["layer_norm", "softmax", "gelu"]):
+                loss = model(
+                    query_input_ids=query_input_ids,
+                    title_input_ids=title_input_ids,
+                    query_token_type_ids=query_token_type_ids,
+                    title_token_type_ids=title_token_type_ids)
+
+            if args.use_amp:
+                scaler.scale(loss).backward()
```
Comment (guoshengCS) [inline examples/semantic_indexing/train_batch_neg.py]: 这里最好是依照grad_scaler文档https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/amp/grad_scaler.py#L72 加个scale_loss作为`scaler.scale(loss)`的结果吧，backward及minimize都使用scale_loss吧，之前BERT中应该是忘记加了，两个的效果应该是一致的，不过感觉使用没有scale的loss表义上不是特别清晰

**Review by guoshengCS — DISMISSED** [review]

File: examples/semantic_indexing/train_batch_neg.py (hunk 1 above)
Comment (tianxin1860) [reply examples/semantic_indexing/train_batch_neg.py]: 已修正，thanks~

**Review by guoshengCS — APPROVED** [review]
